docs/interactions/Application_Commands.md

```diff
@@ -4,8 +4,9 @@ Application commands are commands that an application can register to Discord. T
 
 ## Application Command Object
 
-> info
-> `CHAT_INPUT` command names must be all lowercase matching `^[\w-]{1,32}$`. `USER` and `MESSAGE` commands may be mixed case and can include spaces.
+###### Application Command Naming
+
+`CHAT_INPUT` command names and command option names must be a combination of up to 32 letters, digits, the underscore charater '_' or the hyphen character '-'. If there is a lowercase variant of any of the letters you must use those, characters with no lowercase variants and/or uncased letters are still allowed. `USER` and `MESSAGE` commands may be mixed case and can include spaces.
```

Comment:
Personally, I find this wording is less readable than the existing one. The regular expression is, though not perfectly accurate, far more expressive than the suggested change.

Reply:
The reason I have removed the regex is because it's ambiguous without more information. Regexes are implementation dependent and not unicode aware by default in some engines.
 
This description is more accurate in these cases because, by providing the category of character or the character itself, people can construct matching rules to suit their system. I for one use regex alternatives and this documentation is much more informative. 

Secondly "lowercase matching this regex" is wrong. You can use uppercase characters as long as they have no lower case equivalent (i.e after case folding they do not produce a lower case letter), you can also use any letter which is not classed as a cased letter. I've made this change to be sure there's no interpretation necessary regarding regex engines and to describe some of the nuances of the actual system.

I'm happy to provide more explicit unicode information if you think that would be more helpful.

Reply:
> underscore charater

Typo on `character`.

> If there is a lowercase variant of any of the letters you must use those, characters with no lowercase variants and/or uncased letters are still allowed.

Change `,` to `;` or split to a new sentence.